x-pack/plugins/apm/server/lib/ui_filters/local_ui_filters/get_local_filter_query.ts

```diff
@@ -23,15 +23,17 @@ export const getLocalFilterQuery = ({
   const field = localUIFilters[localUIFilterName];
   const filter = getUiFiltersES(omit(uiFilters, field.name));
 
-  const bucketCountAggregation = projection.body.aggs
+  const projectionSource =
+    projection.body.aggs &&
+    projection.body.aggs[Object.keys(projection.body.aggs)[0]].terms;
+  const bucketCountAggregation = projectionSource
     ? {
         aggs: {
           bucket_count: {
-            cardinality: {
-              field:
-                projection.body.aggs[Object.keys(projection.body.aggs)[0]].terms
-                  .field,
-            },
+            cardinality:
+              'field' in projectionSource
```

Comment:
I don't think we expect a `script` option here. I feel like we can solve this differently. Do you mind if I push a small change to your branch?

Reply:
Feel free!